speedtest/server_test.go

```diff
@@ -195,3 +196,19 @@ func TestTotalDurationCount(t *testing.T) {
 		t.Error("addition in testDurationTotalCount didn't work")
 	}
 }
+
+func TestCityFlag(t *testing.T) {
+	client := New(WithUserConfig(&UserConfig{CityFlag: "yishun"}))
+	servers, err := client.FetchServers()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+	row := 1
+	for _, server := range servers {
+
+		if server.CC != strings.ToUpper(client.config.Location.CC) {
+			t.Error("server country code does not match client country code")
+		}
+		row++
+	}
+}
```

Comment:
```suggestion
func TestCityFlag(t *testing.T) {
	testCC := "YISHUN"
	testData := Servers{
		{CC: "YISHUN"},
		{CC: "TOKYO"},
		{CC: "YISHUN"},
		{CC: "TEST"},
	}

	var tmpServers Servers
	for _, server := range testData {
		if server.CC == strings.ToUpper(testCC) {
			tmpServers = append(tmpServers, server)
		}
	}
	if tmpServers.Len() != 2 && tmpServers[0].CC != testCC {
		t.Fatalf("not match: %s", testCC)
	}
}
```

Reply:
with the suggestion mean that the test doesn't check with actual data for the speedtest server ?
only test the filtering logic ?

Reply:
Yes